core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -636,8 +642,12 @@ class CivilizationInfo {
         val militaryUnit = city.cityConstructions.getConstructableUnits()
                 .filter { !it.unitType.isCivilian() && it.unitType.isLandUnit() }
                 .toList().random()
-        placeUnitNearTile(city.location, militaryUnit.name)
-        addNotification("[${otherCiv.civName}] gave us a [${militaryUnit.name}] as gift near [${city.name}]!", city.location, otherCiv.civName, militaryUnit.name)
+        // placing the unit may fail - in that case stay quiet
+        val placedUnit = placeUnitNearTile(city.location, militaryUnit.name) ?: return
+        // Point to the places mentioned in the message in that order
+        val placedLocation = placedUnit.getTile().position
+        val locations = LocationAction(listOf(getCapital().location, placedLocation, city.location))
```

Comment:
Why the capital? I think just the placed location is needed

Reply:
The capital of the gifting civ - the city state - you don't always remember where they are, even if the gift appears at your closest city. Scouts explore _much_ quicker than your empire expands, right?

Reply:
I see
This is a misunderstanding based on lies.
This function *actually* just gives YOU a unit, and says that the unit came from the city state (this is also an open issue)
Therefore the capital in this case will also be your capital, until we fix the function
So we can change it to otherCiv.getCapital()

Reply:
Ah -
```
                if (!hasFlag(DiplomacyFlags.ProvideMilitaryUnit)) setFlag(DiplomacyFlags.ProvideMilitaryUnit, 20)
```
runs on _my_ civ, so the citystate is the `otherCiv` - and the function should be called `getGiftFrom`, right?

And #3967 is correct because civ1 in `getClosestCities` is me, and it takes city1 -> to fix the issue we would simply take city2 as source and gate through a filter allowing unique differences but not lack tech - would need some changes in that rejection reason function or even more code duplication, so I left it out. We should get the city states their unique units as told by the wiki first.......

Anyway, that was the one I didn't test, of course - found no save with militaristic CS alive. So, changing my decision to present stuff in the order of the message -> push imminent (message says who what near, I think jumps may be better what who near). I'll recheck my saves for an example...